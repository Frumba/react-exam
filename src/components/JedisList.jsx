import React, { PropTypes } from 'react';

const JedisList = (props) => {
  const { jedis } = props;

  return (
    <div className="jedis">
      {jedis.map((jedi) => (
        <div key={jedi.id}>
          Jedi: id: {jedi.id} name: {jedi.name}
        </div>
      ))}
    </div>
  );
};

JedisList.propTypes = {
  jedis: PropTypes.array
};

export default JedisList;
