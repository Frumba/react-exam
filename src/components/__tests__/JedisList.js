import React from 'react';
import renderer from 'react-test-renderer';

import JedisList from '../JedisList';

import dbJSON from '../../../api/db.json';

it('Jedis list empty', () => {
  const tree = renderer.create(<JedisList />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Jedis list with elements', () => {
  const tree = renderer.create(<JedisList jedis={dbJSON.jedis} />).toJSON();
  expect(tree).toMatchSnapshot();
});
