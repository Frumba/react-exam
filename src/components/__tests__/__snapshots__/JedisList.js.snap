// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jedis list empty 1`] = `null`;

exports[`Jedis list with elements 1`] = `
Array [
  <div>
    Jedi: id: 
    1
     name: 
    Qui-Gon Jinn
  </div>,
  <div>
    Jedi: id: 
    2
     name: 
    Mace Windu
  </div>,
  <div>
    Jedi: id: 
    3
     name: 
    Aayla Secura
  </div>,
  <div>
    Jedi: id: 
    4
     name: 
    Yoda
  </div>,
  <div>
    Jedi: id: 
    5
     name: 
    Plo Koon
  </div>,
  <div>
    Jedi: id: 
    6
     name: 
    Luke Skywalker
  </div>,
  <div>
    Jedi: id: 
    7
     name: 
    Adi Gallia
  </div>,
  <div>
    Jedi: id: 
    8
     name: 
    Agen Kolar
  </div>,
  <div>
    Jedi: id: 
    9
     name: 
    Ki-Adi-Mundi
  </div>,
  <div>
    Jedi: id: 
    10
     name: 
    Quinlan Vos
  </div>,
]
`;
